src/commands/deploy.ts

```diff
@@ -115,10 +115,16 @@ export async function deploy(context: IActionContext, confirmDeployment: boolean
         // currentWorkspace is only set if there is one active workspace
         // only check enableScmDoBuildDuringDeploy if currentWorkspace matches the workspace being deployed as a user can "Browse" to a different project
         if (workspaceConfig.get(constants.configurationSettings.showBuildDuringDeployPrompt)) {
-            if (siteConfig.linuxFxVersion && siteConfig.linuxFxVersion.startsWith(constants.runtimes.node) && siteConfig.scmType === 'None' && !(await pathExists(path.join(fsPath, constants.deploymentFileName)))) {
-                // check if web app has node runtime, is being zipdeployed, and if there is no .deployment file
-                // tslint:disable-next-line:no-unsafe-any
-                await node.enableScmDoBuildDuringDeploy(fsPath, constants.runtimes[siteConfig.linuxFxVersion.substring(0, siteConfig.linuxFxVersion.indexOf('|'))], context.properties);
+            //check if node is being zipdeployed and that there is no .deployment file
+            if (siteConfig.linuxFxVersion && siteConfig.scmType === 'None' && !(await pathExists(path.join(fsPath, constants.deploymentFileName)))) {
+                if (siteConfig.linuxFxVersion.startsWith(constants.runtimes.node)) {
+                    // if it is node, prompt the user (as we can break them)
+                    await node.promptScmDoBuildDeploy(fsPath, constants.runtimes[constants.runtimes.node], context.properties);
+                } else if (siteConfig.linuxFxVersion.startsWith(constants.runtimes.python)) {
+                    // if it is a python app, enable it for them as it does not work without build during deploy
+                    await node.enableScmDoBuildDuringDeploy(fsPath, constants.runtimes.python);
```

Comment:
> it does not work without build during deploy

Can you point to docs for this? And will this always be the case? I don't like this for two reasons:
1. It happens on every deploy and users have no way to override the behavior if they don't want it
1. I think it's confusing when tooling changes a user's project without telling them

Reply:
Okay so, I tried creating the venv and then pip install the requirements and deployed that and it worked.  I'll handle it the same way we handle node.js.